spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/ReactiveKafkaProperties.java

```diff
@@ -0,0 +1,341 @@
+
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.consumer.RetriableCommitFailedException;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import reactor.kafka.receiver.KafkaReceiver;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.PropertyMapper;
+import org.springframework.util.CollectionUtils;
+
+/**
+ * Configuration properties for Project Reactor for Apache Kafka.
+ * <p>
+ * Users should refer to Reactor Kafka documentation for complete descriptions of these
+ * properties.
+ *
+ * @author Almog Tavor
+ * @since 2.7.0
+ */
+@ConfigurationProperties(prefix = "spring.reactor.kafka")
+public class ReactiveKafkaProperties {
+
+	/**
+	 * Additional properties, common to producers and consumers, used to configure the
+	 * client.
+	 */
+	private final Map<String, String> properties = new HashMap<>();
+
+	private final Receiver receiver = new Receiver();
+
+	private final Sender sender = new Sender();
+
+	public Map<String, String> getProperties() {
+		return this.properties;
+	}
+
+	public Receiver getReceiver() {
+		return this.receiver;
+	}
+
+	public Sender getSender() {
+		return this.sender;
+	}
+
+	private Map<String, Object> buildCommonProperties() {
+		Map<String, Object> commonProperties = new HashMap<>();
+		if (!CollectionUtils.isEmpty(this.properties)) {
+			commonProperties.putAll(this.properties);
+		}
+		return commonProperties;
+	}
+
+	/**
+	 * Create an initial map of consumer properties from the state of this instance.
+	 * <p>
+	 * This allows you to add additional properties, if necessary, and override the
+	 * default kafkaReceiver bean.
+	 * @return the consumer properties initialized with the customizations defined on this
+	 * instance
+	 */
+	public Map<String, Object> buildReceiverProperties() {
+		Map<String, Object> receiverProperties = buildCommonProperties();
+		receiverProperties.putAll(this.receiver.buildProperties());
+		return receiverProperties;
+	}
+
+	/**
+	 * Create an initial map of producer properties from the state of this instance. This
+	 * allows you to add additional properties, if necessary, and override the default
+	 * kafkaSender bean.
+	 * @return the producer properties initialized with the customizations defined on this
+	 * instance
+	 */
+	public Map<String, Object> buildSenderProperties() {
+		Map<String, Object> senderProperties = buildCommonProperties();
+		senderProperties.putAll(this.sender.buildProperties());
+		return senderProperties;
+	}
+
+	public static class Receiver {
+
+		/**
+		 * Sets the timeout for each {@link KafkaConsumer#poll(Duration)} operation. Since
+		 * the underlying Kafka consumer is not thread-safe, long poll intervals may delay
+		 * commits and other operations invoked using
+		 * {@link KafkaReceiver#doOnConsumer(java.util.function.Function)}. Very short
+		 * timeouts may reduce batching and increase load on the broker.
```

Comment:
The description is these configuration properties must be short and simple.
And it cannot be like a method Javadoc: we must not use any Javadoc tag and style in this descriptions.
This property indeed doesn't set anything.
See docs for more info: https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#features.developing-auto-configuration.custom-starter.configuration-keys